source/server/CMakeLists.txt

```diff
@@ -0,0 +1,25 @@
+add_library(ni_grpc_device_server_core
+  "device_enumerator.cpp"
+  "feature_toggles.cpp"
+  "logging.cpp"
+  "semaphore.cpp"
+  "server_configuration_parser.cpp"
+  "server_security_configuration.cpp"
+  "session_repository.cpp"
+  "session_utilities_service.cpp"
+  "shared_library.cpp"
+  "syscfg_library.cpp")
+
+target_include_directories(ni_grpc_device_server_core
+  PUBLIC ${CMAKE_SOURCE_DIR}/imports/include
+  PUBLIC "../" # HACK should be able to sanitize this, but include statements expect it to be server/<whatever>.h
+)
+
+target_link_libraries(ni_grpc_device_server_core
+  ni_grpc_device_server_core_proto
+  ${_REFLECTION}
+  ${_GRPC_GRPCPP}
+  ${_PROTOBUF_LIBPROTOBUF}
+  ${CMAKE_DL_LIBS}
```

Comment:
Note: I think only the targets that include the driver libraries needs CMAKE_DL_LIBS (at this point that's the server exe and the test exes; eventually they may have their own library/libraries).